jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/StepsRunner.java

```diff
@@ -331,20 +333,22 @@ private void pushBaseImageLayers() {
     ProgressEventDispatcher.Factory childProgressDispatcherFactory =
         Verify.verifyNotNull(rootProgressDispatcher).newChildProducer();
 
-    results.baseImageLayerPushResults =
+    results.builtImagesAndLayerPushResults =
         executorService.submit(
             () -> {
-              List<List<Future<BlobDescriptor>>> pushResults = new ArrayList<>();
-              for (List<Future<PreparedLayer>> baseImageLayers :
-                  results.baseImagesAndLayers.get().values()) {
+              Map<Future<Image>, List<Future<BlobDescriptor>>> pushResults = new HashMap<>();
+
+              for (Map.Entry<Future<Image>, List<Future<PreparedLayer>>> entry :
+                  results.builtImagesAndLayers.get().entrySet()) {
```

Comment:
I don't think this is correct. As the original code, it should depend on `baseImagesAndLayers`. Depending on `builtImages...` means that this code will block until you complete building `Image`s.

Reply:
The logic here is to be able to keep track of `builtimage -> baseImageLayerPushResults` so that we can be able to use `realizeFutures` on the correct `baseImageLayerPushResults` in the `pushImages` step.

The reason why I am depending on `builtImages` here is because inorder to be able to match `builtImage -> baseImageLayerPushResults` you need to have the `builtImage -> baseImageLayer` mapping first which I create in the `buildimage` function.

Reply:
I can understand why it ended up like this, but I think this is not correct for a few reasons. I will just mention one reason that can alone make this not acceptable: 

`results.builtImagesAndLayers` is the result coming from scheduling a thread in `buildImages()`. And calling `results.builtImagesAndLayers.get()` here means you will wait until the thread scheduled in `buildImages()` runs to completion. Conceptually, it means you'll wait until "images" are built. The code will now speak that "in order to be able to push base image layers, I need to build images first." OTOH, previously, this `pushBaseImageLayers()` didn't have any dependency on `buildImages()`. Pushing base image layers can start as soon as we have base image layers. But with this change, you can start pushing only after the thread from `buildImages()` completed. This is a significant departure from the current async dependencies behaviors. Whenever we change the original intention, there should be a clear reason; just to allow making some other parts technically easy to make it work with minimal changes can't be a reason.

Reply:
That makes sense - this implementation does slow down things.I'l revert this to the previous implementation and we can make changes starting from there.